dask/dataframe/dask_expr/_expr.py

```diff
@@ -3804,19 +3806,26 @@ def __le__(self, other):
         return True
 
 
-def determine_column_projection(expr, parent, dependents, additional_columns=None):
+def determine_column_projection(
+    expr: Expr,
+    parent: Expr,
+    dependents: dict[str, Collection[weakref.ref[BaseExpr]]],
+    additional_columns: list | None = None,
+):
```

Comment:
I can't remember mypy's rules offhand (or whether we're using it), but do we need a return type for it to check the body of the function? Maybe not, since you have a `type: ignore` down below, in which case ignore this.

Reply:
not sure. I added an annotation but kept it at object since the output is not easily defined